python/ray/data/aggregate.py

```diff
@@ -57,49 +57,115 @@ def __init__(self):
 class Sum(AggregateFn):
     """Defines sum aggregation."""
 
-    def __init__(self, on: Optional[AggregateOnT] = None):
+    def __init__(self,
+                 on: Optional[AggregateOnT] = None,
+                 ignore_nulls: bool = True):
```

Comment:
I think that true by default will be more user-friendly and seems more standard, e.g. `skipna` default to `True` for Pandas: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.mean.html

Reply:
If we raise an error when null is encountered and `ignore_nulls` is False, then I think users will never want to set it to False? In contrast, when skipna is False, pandas mean returns NaN instead of exception.

I'm just wondering what users want/expect when they set ignore_nulls to False.

Reply:
My thinking was that the user would want to set it to `False` if they're expecting there to be no nulls in the column, in which case they'd want an error to be thrown if an erroneous null is encountered.

>In contrast, when skipna is False, pandas mean returns NaN instead of exception.

We could do this instead of throwing an error, do you think those semantics are preferable? I opted for eagerly throwing an error so the computation could short-circuit, but returning a NaN is probably better UX?

Reply:
I'm actually not sure which is better. Probably it's a question to the real dataset user. Fail fast seems fine to me.

Reply:
I think that returning `None` if `ignore_nulls=False` and a null is encountered might be better than throwing an error; we'll already need to handle `None` propagation if we want to handle the all null case well.

Reply:
ok, let's see what @ericl or other people say. I don't have a strong opinion here.